src/newcomponents/hub/chatForm/ChatFormInputs.js

```diff
@@ -0,0 +1,170 @@
+import api from "@api/api";
+import globalConstants from "@helpers/constants.js";
+import isMobile from "@utils/get_device_type.js";
+import jwtDecode from "jwt-decode";
+import showCustomAlert from "@utils/show_alert";
+import {
+  addMessage,
+  getActiveConversationMessages,
+  removeMessage,
+} from "@store/values/Messages";
+import {
+  getConverastionById,
+  setLastMessageField,
+  updateLastMessageField,
+} from "@store/values/Conversations";
+import { getNetworkState } from "@store/values/NetworkState";
+import { useDispatch, useSelector } from "react-redux";
+import { useEffect, useRef, useState } from "react";
+
+import "@newstyles/hub/chatForm/ChatFormInputs.css";
+
+import { ReactComponent as Attach } from "@newicons/options/Attach.svg";
+import { ReactComponent as Send } from "@newicons/options/Send.svg";
+
+export default function ChatFormInputs({
+  chatMessagesBlockRef,
+  messageInputEl,
+}) {
+  const dispatch = useDispatch();
+
+  const connectState = useSelector(getNetworkState);
+  const userInfo = localStorage.getItem("sessionId")
+    ? jwtDecode(localStorage.getItem("sessionId"))
+    : null;
+
+  const selectedConversation = useSelector(getConverastionById);
+  const selectedCID = selectedConversation?._id;
+
+  const messages = useSelector(getActiveConversationMessages);
+  const filePicker = useRef(null);
+  const [isSendMessageDisable, setIsSendMessageDisable] = useState(false);
+
+  const scrollChatToBottom = () =>
+    chatMessagesBlockRef.current?._infScroll?.scrollIntoView({ block: "end" });
+
+  const sendMessage = async (event) => {
+    event.preventDefault();
+
+    if (!connectState) {
+      showCustomAlert("No internet connection…", "warning");
+      return;
+    }
+
+    const text = messageInputEl.current.value.trim();
+    if (text.length === 0 || isSendMessageDisable) {
+      return;
+    }
+    setIsSendMessageDisable(true);
+    messageInputEl.current.value = "";
+    const mid = userInfo._id + Date.now();
+    let msg = {
+      _id: mid,
+      body: text,
+      from: userInfo._id,
+      t: Date.now(),
+    };
+
+    dispatch(addMessage(msg));
+    dispatch(updateLastMessageField({ cid: selectedCID, msg }));
+
+    const reqData = {
+      mid,
+      text: text,
+      chatId: selectedCID,
+    };
+
+    let response;
```

Comment:
I propose to provide all this logic as a part of `onSubmit` prop, In this case the `ChatFormInputs ` component will be less coupled. Otherwise, `ChatFormInputs ` component knows to much information